joomla/joomla-cms
# [4.0] Fix padding for menu item edit link in Cassiopeia frontend editing

Pull Request for [https://github.com/joomla/joomla-cms/pull/32070#issuecomment-763919602](https://github.com/joomla/joomla-cms/pull/32070#issuecomment-763919602) issue 1.

### Summary of Changes

Remove unwanted padding from menu item edit icons.

### Testing Instructions

1. On a current 4.0-dev branch or latest nightly built with the merged PR #32070 included, install blog sample data if you haven't done that yet, or make sure in some other way to have enough menus at diverse positions to verify the issue and the fix.
2. Enable frontend editing for menus:
![2021-01-23_06](https://user-images.githubusercontent.com/7413183/105577870-8db96f00-5d7c-11eb-9dea-e4bca8559e7a.png)
3. Login to frontend and go to some other page than the start page to get rid of the big image being shown at the top.
4. Check horizontal alignment of menu items (in case of a horizontal menu the first one) in menu modules to other modules' content.
Result: There is some padding before the menu item's edit icon.
5. Log out so there is no frontend editing and check again alignment, check again horizontal alignment of menu items and note the change in vertical module positions.
Result: Without frontend editing, menu items are aligned well horizontally with other content. Thing below the header section moved a bit up compared to frontend editing on.
6. Check the same (steps 4 and 5) for LTR. You can easily change to LTR by editing file `language/en-GB/langmetadata.xml` and changing `<rtl>0</rtl>` to `<rtl>1</rtl>` in the `<metadata>` section.
Result: Not that easy to see due to the other issue with the position of the module edit buttons.
7. Apply the patch of this PR.
8. Repeat steps 3 to 6.
Result: See section "Expected result AFTER applying this Pull Request" below.

### Actual result BEFORE applying this Pull Request

LTR with frontend editing enabled:

![2021-01-23_01](https://user-images.githubusercontent.com/7413183/105577760-cd338b80-5d7b-11eb-924d-f3f302162ff6.png)

The menu items are not aligned as they would be when frontend editing was switched off, see red lines in the screenshot above and green lines in the one below.

LTR without frontend editing:

![2021-01-23_00](https://user-images.githubusercontent.com/7413183/105578447-91e78b80-5d80-11eb-83aa-cfb7437f2de7.png)

The 2 screenshots have been made with the same window size. As you can see by the articles' texts at the bottom, the vertical space needed by the horizontal menu in the header changes when frontend editing is on, so everything below the header moves a bit down.

### Expected result AFTER applying this Pull Request

LTR with frontend editing enabled:

![2021-01-23_02](https://user-images.githubusercontent.com/7413183/105578102-2b616e00-5d7e-11eb-86e3-2800f9bdf8d7.png)

Compared to screenshot 2 in the previous section, the vertical positions of things below the header area haven't changed.

The horizontal alignment of menu item edit icons to menu titles or other content is the same as the horizontal alignment of menu item titles without frontend editing e.g. when logged off.

For RTL, the results are harder to verify due to the other issue with position of the module edit buttons.

### Additional information

The underline of the menu item edit link in the header section has to be fixed with another PR.

![2021-01-23_07](https://user-images.githubusercontent.com/7413183/105579263-c3af2100-5d85-11eb-8f62-f59b9292cc2d.png)

I've tried a bit when doing this PR here but did not really find a nice way. It should be removed completely for the edit icon in _header.scss (and possibly also _footer.scss).

### Documentation Changes Required

None.
